library/forms/lib/formModule/formManagers/utils/autoIncrementInitializer.ts

```diff
@@ -0,0 +1,90 @@
+import {AutoIncrementFieldRef, AutoIncrementService} from '../../incrementer';
+import {FaimsForm, FaimsFormData} from '../../types';
+
+export interface InitializeAutoIncrementFieldsParams {
+  /** The form instance to update */
+  form: FaimsForm;
+  /** The form ID to filter auto-increment fields */
+  formId: string;
+  /** The auto-increment service */
+  incrementerService: AutoIncrementService;
+  /** Initial data - fields with existing values won't be overwritten */
+  initialData?: FaimsFormData;
+  /** Called if any field couldn't get a value (ranges exhausted/not configured) */
+  onMissingRanges?: (fieldRefs: AutoIncrementFieldRef[]) => void;
+}
+
+export interface InitializeAutoIncrementFieldsResult {
+  /** Fields that were successfully initialized */
+  initialized: AutoIncrementFieldRef[];
+  /** Fields that couldn't get values (no ranges or exhausted) */
+  missingRanges: AutoIncrementFieldRef[];
+  /** Fields that were skipped because they already had values */
+  skipped: AutoIncrementFieldRef[];
+}
+
+/**
+ * Initializes auto-increment fields on a form by fetching next values
+ * from the incrementer service and setting them on the form.
+ *
+ * Only sets values for fields that don't already have a value in initialData.
+ */
+export async function initializeAutoIncrementFields({
+  form,
+  formId,
+  incrementerService,
+  initialData,
+  onMissingRanges,
```

Comment:
this is called onIssue in the useIncrementerService hook - would be good to be consistent